lib/ansible/modules/package_facts.py

```diff
@@ -259,12 +259,38 @@
 
 
 ALIASES = {
-    'rpm': ['dnf', 'dnf5', 'yum' , 'zypper'],
+    'dnf': ['dnf', 'dnf5'],
+    'rpm': ['yum' , 'zypper'],
     'pkg': ['pkg5', 'pkgng'],
     'pkg_info': ['openbsd_pkg'],
 }
 
 
+class DNF(RespawningLibMgr):
+    LIB = "dnf"
+    CLI_BINARIES = ['dnf']
```

Comment:
You may want this to be `dnf-3`. Which oddly enough is actually `dnf4`:

```
[root@872d03d76c75 /]# dnf-3 --version
4.14.0
  Installed: dnf-0:4.14.0-18.el9.noarch at Mon Oct  7 04:10:09 2024
  Built    : builder@centos.org at Tue Oct  1 14:26:18 2024

  Installed: rpm-0:4.16.1.3-34.el9.aarch64 at Mon Oct  7 04:10:07 2024
  Built    : builder@centos.org at Thu Aug 15 09:02:12 2024
[root@872d03d76c75 /]# ls -al /usr/bin/dnf
lrwxrwxrwx 1 root root 5 Oct  1 14:26 /usr/bin/dnf -> dnf-3
```

Reply:
aka reaons why i stuck with rpm ....